app/models/history.rb

```diff
@@ -463,30 +466,42 @@ def self.clear_follow_up_flag(history)
   end
 
   private_class_method def self.compose_message(history, field)
-    creator = history[:household_status] == :patient ? 'User' : 'System'
+    creator = history[:household_status].nil? ? 'System' : 'User'
     verb = field[:new_value].blank? ? 'cleared' : 'changed'
     from_text = field[:old_value].blank? ? 'blank' : "\"#{field[:old_value]}\""
     to_text = field[:new_value].blank? ? 'blank' : "\"#{field[:new_value]}\""
 
     comment = "#{creator} #{verb} #{field[:name]} from #{from_text} to #{to_text}"
-    comment += compose_explanation(history, field)
+    comment += compose_explanation(history)
     comment += history[:note] unless history[:note].blank?
     comment += '.'
     comment += " Reason: #{history[:reason]}" unless history[:reason].blank?
     comment
   end
 
-  private_class_method def self.compose_explanation(history, field)
-    if history[:household_status] == :patient && history[:propagation] == :group
-      " and chose to update this #{field[:type]} for all household members"
-    elsif history[:household_status] == :patient && history[:propagation] == :group_cm
-      " and chose to update this #{field[:type]} for household members under continuous exposure"
-    elsif history[:household_status] != :patient && history[:propagation] == :group
-      " because User updated #{field[:name]} for another member in this monitoree's household and chose to update this
-        #{field[:type].nil? ? 'field' : field[:type]} for all household members"
-    elsif history[:household_status] != :patient && history[:propagation] == :group_cm
-      " because User updated #{field[:name]} for another member in this monitoree's household and chose to update this
-        #{field[:type].nil? ? 'field' : field[:type]} for household members under continuous exposure"
+  private_class_method def self.compose_explanation(history)
+    if history[:household_status] == history[:patient].id && history[:propagation] == :group
+      ' and applied that change to all household members'
+    elsif history[:household_status] == history[:patient].id && history[:propagation] == :group_cm
+      ' and applied that change to household members under continuous exposure'
+    elsif history[:household_status] == history[:patient].responder_id && history[:propagation] == :group
+      " by making that change for monitoree's head of household (Sara Alert ID: #{history[:household_status]})
```

Comment:
has all of this wording been verified with the PO team?

```suggestion
      " by making that change for this monitoree's head of household (Sara Alert ID: #{history[:household_status]})
```

Reply:
The PO team reviewed my original text and said it was overall fine but needed to be shorter after the (Sara Alert ID: adfadf) part, and haven't yet seen the shortened text.

Reply:
@timwongj PO team has signed off on the version I just pushed!